app.go

```diff
@@ -308,11 +336,48 @@ func New(opts ...Option) *App {
 
 	if err := app.executeInvokes(); err != nil {
 		app.err = err
+		var b bytes.Buffer
+
+		if dig.CanVisualizeError(err) {
+			dig.Visualize(app.container, &b, dig.VisualizeError(err))
+		}
+
+		app.errorHooks.onError(wrappedErr{
+			graph: b.String(),
+			err:   err,
+		})
 	}
 
 	return app
 }
 
+type wrappedErr struct {
+	graph string
+	err   error
+}
+
+type isWrapped interface {
+	Graph() string
+}
```

Comment:
We can drop this interface and cast the `err` directly to `wrappedErr` (whatever its name ends up being).

```
if e, ok := err.(wrappedErr); ok {
  return e.graph
}
```

Reply:
The cast to interface is needed for the tests to work, since I can't create a `wrappedErr` in the tests (they're in the `fx_test` package)